In [2]:
import pandas as pd

from os import path
import pathlib ##Getting to work in Notebooks
import glob

In [5]:
## Access folder where csv data is stored

## Code from previous project; altered to work for jupyter notebooks
## For code outside of notebooks, use: 
##     basepath = path.dirname(__file__)
##     filepath = path.abspath(path.join(basepath, "data"))

basepath = pathlib.Path().resolve()
filepath = path.abspath(path.join(basepath,"data"))


## https://stackoverflow.com/questions/20906474
## /import-multiple-csv-files-into-pandas-and-concatenate-into-one-dataframe
## Pulling from csvs when they're in a different location

files = glob.glob(filepath + "/*.csv")

sets = []

for filename in files:
    f = pd.read_csv(filename, index_col = None, header = 0)
    sets.append(f)
    
df = pd.concat(sets, axis = 0, ignore_index = True)

In [6]:
df.head(2)

backers_count                                              blurb  \
0             23  Ginger KICK! is back for the holidays with ama...   
1            322  Let's go to the movies on the L.A. River! Help...   

                                            category  \
0  {"id":313,"name":"Small Batch","slug":"food/sm...   
1  {"id":298,"name":"Movie Theaters","slug":"film...   

   converted_pledged_amount country  created_at  \
0                       825      US  1509883503   
1                     10545      US  1361914696   

                                             creator currency currency_symbol  \
0  {"id":990359968,"name":"Danielle Ackley-McPhai...      USD               $   
1  {"id":860373786,"name":"L.A. River Revitalizat...      USD               $   

   currency_trailing_code  ...                            slug  \
0                    True  ...       ginger-kick-holiday-cheer   
1                    True  ...  la-river-bike-in-movie-theater   

                                          source_url  spotlight staff_pick  \
0  https://www.kickstarter.com/discover/categorie...       True      False   
1  https://www.kickstarter.com/discover/categorie...       True       True   

        state  state_changed_at  static_usd_rate  \
0  successful        1510518809              1.0   
1  successful        1365700816              1.0   

                                                urls  usd_pledged  usd_type  
0  {"web":{"project":"https://www.kickstarter.com...        825.0  domestic  
1  {"web":{"project":"https://www.kickstarter.com...      10545.0  domestic  

[2 rows x 37 columns]

In [7]:
df.describe()

backers_count  converted_pledged_amount    created_at      deadline  \
count  210088.000000              2.100880e+05  2.100880e+05  2.100880e+05   
mean      145.346417              1.285993e+04  1.457802e+09  1.464764e+09   
std       890.106978              9.002384e+04  6.419645e+07  6.386306e+07   
min         0.000000              0.000000e+00  1.240366e+09  1.242468e+09   
25%         4.000000              1.090000e+02  1.414097e+09  1.421827e+09   
50%        27.000000              1.549000e+03  1.459286e+09  1.466077e+09   
75%        89.000000              6.591000e+03  1.514907e+09  1.522030e+09   
max    105857.000000              1.046949e+07  1.555533e+09  1.560744e+09   

             fx_rate          goal            id   launched_at       pledged  \
count  210088.000000  2.100880e+05  2.100880e+05  2.100880e+05  2.100880e+05   
mean        0.992695  4.984399e+04  1.073280e+09  1.461933e+09  1.838593e+04   
std         0.209140  1.208255e+06  6.198001e+08  6.389101e+07  3.060742e+05   
min         0.008932  1.000000e-02  8.624000e+03  1.240674e+09  0.000000e+00   
25%         1.000000  1.500000e+03  5.353658e+08  1.418785e+09  1.110000e+02   
50%         1.000000  5.000000e+03  1.074189e+09  1.463243e+09  1.566000e+03   
75%         1.000000  1.500000e+04  1.610073e+09  1.519245e+09  6.980000e+03   
max         1.743276  1.000000e+08  2.147476e+09  1.555563e+09  8.103074e+07   

       state_changed_at  static_usd_rate   usd_pledged  
count      2.100880e+05    210088.000000  2.100880e+05  
mean       1.464569e+09         1.010243  1.286061e+04  
std        6.371158e+07         0.232424  9.003800e+04  
min        1.242468e+09         0.008771  0.000000e+00  
25%        1.421640e+09         1.000000  1.100000e+02  
50%        1.466013e+09         1.000000  1.550000e+03  
75%        1.521988e+09         1.000000  6.598000e+03  
max        1.555563e+09         1.716408  1.046949e+07

## Exploration

There are 210,088 rows of backer data pulled from https://webrobots.io/kickstarter-datasets/
All quantitative fields look to contain all information in some form since their respective counts are 210088. 



Looking at the backers_count field, it's interesting that the 75th percentile is at 89 while the max is at 105857. Additionally, the mean is at 145. With the mean being higher than the median, we can tell that backers_count is right skewed. This makes sense with Kickstarters and how their projects work. Incredibly popular projects will get way more than their goal and have a higher reach. 


The fields created_at, deadline, id, launched_at, state_changed_at look to be different data types than the ones presented by .describe()

In [10]:
df.head()[['created_at', 'deadline', 'launched_at', 'state_changed_at']]

created_at    deadline  launched_at  state_changed_at
0  1509883503  1510518804   1509914004        1510518809
1  1361914696  1365700816   1363108816        1365700816
2  1549877514  1554903476   1552657076        1554903476
3  1502707276  1505921026   1503329026        1505921027
4  1513448977  1534015191   1532805591        1534015191

It looks like these are datetime fields and the dataset is storing them in epoch time. 

In [20]:
# We can import the time module and use either the gmtime or localtime
# to converter epoch to the format we'd like.

import time

epochtime = df.iloc[0]['created_at']

struct = time.gmtime(epochtime)
time_string = time.strftime("%Y-%m-%d, %H:%M:%S", struct)
print(epochtime)
print(time_string)

1509883503
2017-11-05, 12:05:03


In [23]:
def epoch_to_date(epoch):
    struct = time.gmtime(epoch)
    date_string = time.strftime("%Y-%m-%d", struct)
    return date_string
def epoch_to_time(epoch):
    struct = time.gmtime(epoch)
    time_string = time.strftime("%H:%M:%S", struct)
    return time_string

In [27]:
df['created_at_date'] = df['created_at'].apply(epoch_to_date)
df['created_at_time'] = df['created_at'].apply(epoch_to_time)

created_at created_at_date
0  1509883503      2017-11-05
1  1361914696      2013-02-26
2  1549877514      2019-02-11
3  1502707276      2017-08-14
4  1513448977      2017-12-16